hypershift-operator/controllers/nodepool/token.go

```diff
@@ -206,6 +206,12 @@ func setExpirationTimestampOnToken(ctx context.Context, c client.Client, tokenSe
 		now = time.Now
 	}
 
+	// there's no need to set the expiration timestamp annotation again if already set
```

Comment:
Thanks! please let's add test coverage for this in TestSetExpirationTimestampOnToken / cleanupOutdated

Reply:
Added test cases in https://github.com/openshift/hypershift/pull/5942/commits/7a3995a2660f1da3c47202fcc5030d3eb9d75c86
Let me know if something looks incorrect